src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -524,7 +524,24 @@ internal IndexCreationOptions GetIndexCreationOptions(object indexDefinition, In
             return IndexCreationOptions.Update;
         }
 
-        private void ValidateIndexName(string name, bool isStatic)
+        public static bool IsValidResourceName(string name, bool isStatic, out string errorMessage)
+        {
+            errorMessage = null;
+            
+            try
+            {
+               ValidateIndexName(name, isStatic);
+            }
+            catch (Exception e)
+            {
+                errorMessage = e.Message.Split('\n')[0];
```

Comment:
Because e.message contains:

"Index name 'abc?' is not permitted. Only letters, digits and characters that match regex '([_\/\-\.]+)' are allowed. *\r\n* Parameter name: name"